connection.go

```diff
@@ -48,6 +49,22 @@ type mysqlConn struct {
 	finished chan<- struct{}
 	canceled atomicError // set non-nil if conn is canceled
 	closed   atomicBool  // set when conn is closed, before closech is closed
+
+	// for killing query after timeout
+	id int
+	d MySQLDriver
+	dsn string
+}
+
+func (mc *mysqlConn) kill() error {
+	killConn, killErr := mc.d.Open(mc.dsn)
+	if killErr != nil {
+		return killErr
+	}
+	defer killConn.Close()
+	query := fmt.Sprintf("KILL QUERY %d", mc.id)
+	_, err := killConn.(*mysqlConn).Exec(query, []driver.Value{})
```

Comment:
the same variable killErr may be used here.

Reply:
Fixed